infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -102,6 +107,8 @@
 
     private ExecutorService executor;
 
+    private final ReadWriteLock resetLock = new ReentrantReadWriteLock();
```

Comment:
Could we have some comments on this read-write lock? Like how it should be used in the future?
And looks like this PR is pretty different from #2616 , is this PR a subset?

Reply:
comments - done.

This is a `small` fix for 0.3.0.2, yes it's different.